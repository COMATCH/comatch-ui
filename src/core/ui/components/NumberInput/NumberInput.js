import React from 'react';
import classNames from 'classnames';

import { TextInput } from '../TextInput';

import './NumberInput.scss';

/* eslint-disable react/prop-types */
export const NumberInput = ({ type, hideArrows, ...props }) => (
    <TextInput type="number" className={classNames('NumberInput', { 'hide-arrows': hideArrows })} {...props} />
);
