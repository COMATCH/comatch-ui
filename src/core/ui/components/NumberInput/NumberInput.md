## NumberInput

This component does not have any props.

