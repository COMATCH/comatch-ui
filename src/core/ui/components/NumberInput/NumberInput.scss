.NumberInput {
    &.hide-arrows {
        /* Hide up/down arrows on Chrome and Safari */
        input[type="number"]::-webkit-outer-spin-button,
        input[type="number"]::-webkit-inner-spin-button {
            -webkit-appearance: none;
            margin: 0;
        }
        /* Hide up/down arrows on Firefox */
        input[type="number"] {
            -moz-appearance: textfield;
        }
    }
}
