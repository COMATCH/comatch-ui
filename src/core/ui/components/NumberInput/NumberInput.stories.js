/* eslint-disable max-len */
import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { NumberInput } from './NumberInput';

storiesOf('NumberInput', module)
    .add('with basic style', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')} />`}</Highlight>
        </>
    ))
    .add('with placeholder', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} placeholder="Placeholder text" />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')}  placeholder="Placeholder text" />`}</Highlight>
        </>
    ))
    .add('with value', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} value="Value text" />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')}  value="Value text" />`}</Highlight>
        </>
    ))
    .add('with label', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} label="Label text" />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')} label="Label text" />`}</Highlight>
        </>
    ))
    .add('with error', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} label="Label text" inputError="Error text" />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')} label="Label text" inputError="Error text" />`}</Highlight>
        </>
    ))
    .add('hiding Chrome/Safari/Firefox up and down arrows', () => (
        <>
            <NumberInput name="field" onChange={action('changed')} hideArrows />
            <Highlight className="html">{`<NumberInput name="field" onChange={action('changed')} hideArrows />`}</Highlight>
        </>
    ));
