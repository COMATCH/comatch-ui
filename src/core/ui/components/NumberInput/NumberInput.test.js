import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { lorem } from 'faker';
import noop from 'lodash/noop';
import '../../../config/tests/setup';

import { NumberInput } from './NumberInput';

describe('NumberInput tests', () => {
    const REQUIRED_PROPS = {
        name: 'text-area-input',
        type: jest.fn(),
        onChange: jest.fn(),
    };

    const OPTIONAL_PROPS = {
        className: lorem.word(),
        defaultValue: lorem.word(),
        id: lorem.word(),
        inputError: 'Testing InputError Message',
        label: 'Testing Label',
        name: lorem.word(),
        placeholder: lorem.word(),
        value: lorem.word(),
        disabled: false,
        required: true,
        autoComplete: 'on',
        display: 'inline',
        type: 123456789,
        maxLength: 100,
        icon: <i />,
        onBlur: jest.fn(),
        onClick: jest.fn(),
        onFocus: jest.fn(),
    };

    describe('given render the default NumberInput component', () => {
        it('should create and compare snapshot', () => {
            const navigationInput = renderer.create(<NumberInput {...REQUIRED_PROPS} />).toJSON();
            expect(navigationInput).toMatchSnapshot();
        });

        it('should render the default NumberInput without crashing', () => {
            const navigationInput = renderer.create(<NumberInput {...REQUIRED_PROPS} />).toJSON();
            expect(navigationInput).toBeTruthy();
        });
    });

    describe('given only required props', () => {
        it('should render with certain props', () => {
            const props = {
                ...REQUIRED_PROPS,
            };
            const numberInputComponent = shallow(<NumberInput {...props} />);
            // Required props:
            expect(numberInputComponent.prop('name')).toEqual(props.name);
            expect(numberInputComponent.prop('onChange')).toEqual(props.onChange);
            // Defaults:
            expect(numberInputComponent.prop('autoComplete')).toEqual(null);
            expect(numberInputComponent.prop('className')).toEqual('NumberInput');
            expect(numberInputComponent.prop('defaultValue')).toEqual('');
            expect(numberInputComponent.prop('disabled')).toEqual(null);
            expect(numberInputComponent.prop('display')).toEqual('block');
            expect(numberInputComponent.prop('icon')).toEqual(null);
            expect(numberInputComponent.prop('id')).toEqual(null);
            expect(numberInputComponent.prop('inputError')).toEqual('');
            expect(numberInputComponent.prop('label')).toEqual('');
            expect(numberInputComponent.prop('maxLength')).toEqual(null);
            expect(numberInputComponent.prop('onBlur')).toEqual(noop);
            expect(numberInputComponent.prop('onClick')).toEqual(noop);
            expect(numberInputComponent.prop('onFocus')).toEqual(noop);
            expect(numberInputComponent.prop('placeholder')).toEqual('');
            expect(numberInputComponent.prop('required')).toEqual(false);
            expect(numberInputComponent.prop('type')).toEqual('number');
            expect(numberInputComponent.prop('value')).toEqual('');
        });
    });
    describe('given required and all optional props', () => {
        it('should render with certain props', () => {
            const props = {
                ...REQUIRED_PROPS,
                ...OPTIONAL_PROPS,
            };
            const textInputComponent = mount(<NumberInput {...props} />);
            // Required Props:
            expect(textInputComponent.prop('name')).toEqual(props.name);
            expect(textInputComponent.prop('onChange')).toEqual(props.onChange);
            // Optional Props:
            expect(textInputComponent.prop('className')).toEqual(props.className);
            expect(textInputComponent.prop('defaultValue')).toEqual(props.defaultValue);
            expect(textInputComponent.prop('id')).toEqual(props.id);
            expect(textInputComponent.prop('inputError')).toEqual(props.inputError);
            expect(textInputComponent.prop('label')).toEqual(props.label);
            expect(textInputComponent.prop('name')).toEqual(props.name);
            expect(textInputComponent.prop('placeholder')).toEqual(props.placeholder);
            expect(textInputComponent.prop('value')).toEqual(props.value);
            expect(textInputComponent.prop('disabled')).toEqual(props.disabled);
            expect(textInputComponent.prop('required')).toEqual(props.required);
            expect(textInputComponent.prop('autoComplete')).toEqual(props.autoComplete);
            expect(textInputComponent.prop('display')).toEqual(props.display);
            expect(textInputComponent.prop('type')).toEqual(props.type);
            expect(textInputComponent.prop('maxLength')).toEqual(props.maxLength);
            expect(textInputComponent.prop('icon')).toEqual(props.icon);
            expect(textInputComponent.prop('onBlur')).toEqual(props.onBlur);
            expect(textInputComponent.prop('onClick')).toEqual(props.onClick);
            expect(textInputComponent.prop('onFocus')).toEqual(props.onFocus);
        });
    });
    describe('given required props with inputError and label pops', () => {
        it('should render correctly with an InputError', () => {
            const inputWithInputError = <NumberInput {...REQUIRED_PROPS} inputError={OPTIONAL_PROPS.inputError} />;
            const numberInputComponent = mount(inputWithInputError);
            expect(
                numberInputComponent
                    .find('.InputError')
                    .at(0)
                    .text(),
            ).toBe('Testing InputError Message');
        });

        it('should render correctly with a Label', () => {
            const label = <NumberInput {...REQUIRED_PROPS} label={OPTIONAL_PROPS.label} />;
            const numberInputComponent = mount(label);
            expect(
                numberInputComponent
                    .find('.InputLabel')
                    .at(0)
                    .text(),
            ).toBe('Testing Label');
        });
    });
});
