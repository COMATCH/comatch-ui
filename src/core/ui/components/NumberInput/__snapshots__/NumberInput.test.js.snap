// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput tests given render the default NumberInput component should create and compare snapshot 1`] = `
<div
  className="Input TextInput NumberInput block"
  id={null}
>
  
  <input
    autoComplete={null}
    className="TextInput__input"
    disabled={null}
    max={null}
    maxLength={null}
    min={null}
    name="text-area-input"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onClick={[Function]}
    onFocus={[Function]}
    placeholder=""
    type="number"
    value=""
  />
  
</div>
`;
